core/src/main/java/com/microsoft/applicationinsights/channel/concrete/inprocess/PartialSuccessHandler.java

```diff
@@ -0,0 +1,146 @@
+package com.microsoft.applicationinsights.channel.concrete.inprocess;
+
+import java.io.BufferedReader;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.zip.GZIPInputStream;
+
+import org.apache.http.HttpStatus;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.common.base.Optional;
+import com.microsoft.applicationinsights.internal.channel.TransmissionHandler;
+import com.microsoft.applicationinsights.internal.channel.TransmissionHandlerArgs;
+import com.microsoft.applicationinsights.internal.channel.common.GzipTelemetrySerializer;
+import com.microsoft.applicationinsights.internal.channel.common.Transmission;
+import com.microsoft.applicationinsights.internal.channel.common.TransmissionPolicyManager;
+import com.microsoft.applicationinsights.internal.logger.InternalLogger;
+
+public class PartialSuccessHandler implements TransmissionHandler {
+
+	public PartialSuccessHandler(TransmissionPolicyManager policy) {
+
+	}
+
+	@Override
+	public void onTransmissionSent(TransmissionHandlerArgs args) {
+		validateTransmissionAndSend(args);
+
+	}
+
+	public boolean validateTransmissionAndSend(TransmissionHandlerArgs args) {
+		if (args.getTransmission() != null && args.getTransmissionDispatcher() != null) {
+			switch (args.getResponseCode()) {
+			case HttpStatus.SC_PARTIAL_CONTENT:
+				BackendResponse beR = getBackendResponse(args.getResponseBody());
```

Comment:
Nit picking: Please use a more descriptive name instead of beR. "backendResponse" would be fine.

Reply:
Agreed, fixed.